estimagic/differentiation/finite_differences.py

```diff
@@ -0,0 +1,98 @@
+"""Finite difference formulae for jacobians and hessians.
```

Comment:
What about making the three functions one with a keyword argument `differences` which can be `forward`, etc.. For me the difference between the functions would be more obvious.

Reply:
Yes, much better. Also removes the duplication in the docstrings. Thanks!